packages/isc/sandbox_interface.go

```diff
@@ -127,6 +127,10 @@ type Sandbox interface {
 	Privileged() Privileged
 }
 
+type Event interface {
+	Encode() []byte
+}
```

Comment:
I suggest:

```suggestion
type Event interface {
    Topic() Hname
	Payload() []byte
}

func EncodeEvent(Event e) []byte {
    return append(e.Topic().Bytes(), e.Topic.Payload()...)
}
```

This way me make it explicit that all events have a "topic" (that could be anything, we just have to make sure not to use the same topic for 2 different events), and that the first 4 bytes are the topic.

So for example:

```go
type FoundryCreateNewEvent struct {
	SerialNumber uint32
}

func (e *FoundryCreateNewEvent) Topic() isc.Hname {
	return isc.Hn("FoundryCreateNewEvent")
}

func (e *FoundryCreateNewEvent) Payload() []byte {
	return codec.MustEncodeUint32(e.SerialNumber)
}
```

Reply:
I make the interface as a single one

```
type Event interface {
	Topic() []byte
	Payload() []byte
	Encode() []byte
}
```

Reply:
I will separate the Encode() from the interface, but the length of Topic() I am still discussing with @BugFreeSoftware for a comprehensive solution

Reply:
Can you clarify?

I didn't think too much about the decoding because the events are something that we "fire and forget". If a client wants to decode it, it would first inspect the topic, and then based on that decode the payload using the proper Event subtype.

So maybe the Event interface should have a `DecodePayload(payload []byte)`.

Reply:
Sorry I am still confused about why the following interface defeat the purpose of standardize the format.

```
type Event interface {
	Topic() []byte
	Payload() []byte
	Encode() []byte
}
```


The encoded byte slice will still follow the similar format. 

Putting them all together maybe easy to unify them? So have a `DecodePayload(payload []byte)` in the Event interface is good for me

Reply:
It's about not repeating the same logic N times if you have N different events. If `Encode() []byte` is part of the interface then you have to implement tha same thing N times.

Reply:
Oh true. Them I think I want to make the interface as following

```
type Event interface {
	Topic() []byte
	Payload() []byte
        DecodePayload(payload []byte)
}
```

And have two public function 
```
func Encode(e Event) []byte
func Decode(payload []byte) Event
```

Reply:
BTW how can we get iotago.Address from string?

Reply:
`iotago.ParseBech32()`